recipes/LibriSpeech/librispeech_prepare.py

```diff
@@ -454,3 +468,136 @@ def check_librispeech_folders(data_folder, splits):
+        Also download bigger 4grams model
+    """
+    os.makedirs(destination, exist_ok=True)
+    download_file(
+        "https://www.dropbox.com/scl/fi/3fkkdlliavhveb5n3nsow/3gram_lm.arpa?rlkey=jgdrluppfut1pjminf3l3y106&dl=1",
+        os.path.join(destination, "3-gram_sb.arpa"),
+    )
+    if rescoring_lm:
+        download_file(
+            "https://www.dropbox.com/scl/fi/roz46ee0ah2lvy5csno4z/4gram_lm.arpa?rlkey=2wt8ozb1mqgde9h9n9rp2yppz&dl=1",
+            os.path.join(destination, "4-gram_sb.arpa"),
+        )
+
+
+def download_file_and_extract(link, destination):
+    """Download link and unpack it.
+
+    Arguments
+    ---------
+    link : str
+        File to download
+    destination : str
+        Place to put result.
+    """
+    download_file(link, destination)
+    out = destination.replace(".gz", "")
+    if os.path.exists(out):
+        logger.debug("Skipping, already downloaded")
+        return
+    with gzip.open(destination, "rb") as f_in:
+        with open(out, "wb") as f_out:
+            shutil.copyfileobj(f_in, f_out)
+
+
+def dataprep_lm_training(
+    lm_dir, output_arpa, csv_files, external_lm_corpus, vocab_file
+):
+    """Prepare lm txt corpus file for lm training with kenlm (https://github.com/kpu/kenlm)
+    Does nothing if output_arpa exists.
+    Else display to the user how to use kenlm in command line, then exit
+    (return code 1), the user has to run the command manually.
+    Instruction on how to compile kenlm (lmplz binary) is available in the
+    above link.
+
+    Arguments
+    ---------
+    lm_dir : str
+        Path to where to store txt corpus
+    output_arpa : str
+        File to write arpa lm
+    csv_files : List[str]
+        CSV files to use to increase lm txt corpus
+    external_lm_corpus : List[str]
+        (Big) text dataset corpus
+    vocab_file : str
+       N-grams that contain vocabulary items not in this file be pruned.
+    """
+    if not os.path.exists(output_arpa):
+        download_librispeech_lm_training_text(lm_dir)
+        column_text_key = "wrd"  # defined in librispeech_prepare.py
+        lm_corpus = os.path.join(lm_dir, "libri_lm_corpus.txt")
+        line_seen = set()
+        with open(lm_corpus, "w") as corpus:
+            for file in csv_files:
+                for line in get_list_from_csv(file, column_text_key):
+                    corpus.write(line + "\n")
+                    line_seen.add(line + "\n")
+            for file in external_lm_corpus:
+                with open(file) as f:
+                    for line in f:
+                        if line not in line_seen:
+                            corpus.write(line)
+        logger.critical(
+            "RUN the following kenlm command to build a 3-gram arpa LM (https://github.com/kpu/kenlm):"
+        )
+        logger.critical(
+            f"$ lmplz -o 3 --prune 0 1 2 --limit_vocab_file {vocab_file} < {lm_corpus}| sed  '1,20s/<unk>/<UNK>/1' > {output_arpa}"
+        )
+        sys.exit(1)
```

Comment:
TBH, as I explained in previous comments, it would be better to trigger this method in a train.py inside of LibriSpeech/LM/. So that you don't have to sys.exit(1).